recipes/agent.rb

```diff
@@ -40,13 +38,12 @@
   block do
     true
   end
-  Array(node['zabbix']['agent']['service_state']).each do |action|
-    if node['platform_family'] == 'windows'
-      notifies :run, 'execute[install_zabbix_agentd]'
-      notifies :run, 'execute[start_zabbix_agentd]'
-      notifies :run, 'execute[config_firewall]'
-    else
-      notifies :restart, 'service[zabbix_agentd]'
-    end
+  case node['platform_family']
+  when 'windows'
```

Comment:
As I mentioned in [this comment](https://github.com/paxos-bankchain/zabbix/pull/5#discussion_r198286971), we are triggering notifications based on an always-true ruby block. You should instead either "execute" or "not execute" a resource.

If these resources always execute, then these things would not be required. In the notified resource you could just do

```ruby
service "Zabbix Agent" do
  # ...
  action [:enable, restart]
end
```

except, wait, that's a restart! So we're not making anything idempotent - we restart it each time, even if the host is unchanged.